### PR TITLE
feat(common): explain why width/height is required

### PR DESCRIPTION
**Pull request by khempenius**
Update error message text to explain why width & height are required
attributes.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts [hunk 1]
```diff
@@ -507,13 +506,19 @@ function assertNoImageDistortion(
 }
 
 // Verifies that a specified input is set.
-function assertRequiredNumberInput(dir: NgOptimizedImage, inputValue: unknown, inputName: string) {
-  if (typeof inputValue === 'undefined') {
+function assertNonEmptyWidthAndHeight(dir: NgOptimizedImage) {
+  let missingAttributes = [] if (dir.width === undefined) missingAttributes.push('width');
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:510]:
```suggestion
  let missingAttributes = [];
  if (dir.width === undefined) missingAttributes.push('width');
```

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts [hunk 2]
```diff
@@ -507,13 +506,19 @@ function assertNoImageDistortion(
...
+  if (dir.height === undefined) missingAttributes.push('height');
+  if (missingAttributes.length > 0) {
     throw new RuntimeError(
         RuntimeErrorCode.REQUIRED_INPUT_MISSING,
-        `${imgDirectiveDetails(dir.rawSrc)} has detected that the required \`${inputName}\` ` +
-            `attribute is missing. Please specify the \`${inputName}\` attribute ` +
-            `on the mentioned element.`);
+        `${
+            imgDirectiveDetails(
+                dir.rawSrc)} has detected that these required attributes are missing:` +
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:517]: nit: shortening the string should fix the formatting:
```suggestion
        `${imgDirectiveDetails(dir.rawSrc)} has detected that these required attributes ` +
         // ...the rest of the string...
```

**Review by AndrewKushnir — COMMENTED** [review]
@khempenius the error message looks great 👍 Just left a couple comments.

File: packages/common/test/directives/ng_optimized_image_spec.ts [hunk 3]
```diff
@@ -211,9 +211,9 @@ describe('Image directive', () => {
               `NG0${
                   RuntimeErrorCode
                       .REQUIRED_INPUT_MISSING}: The NgOptimizedImage directive (activated on an <img> ` +
-              'element with the `rawSrc="img.png"`) has detected that the required ' +
-              '`height` attribute is missing. Please specify the `height` attribute ' +
-              'on the mentioned element.');
+              'element with the `rawSrc="img.png"`) has detected that these required attributes are missing:' +
+              '`height`. Including "width" and "height" attributes will prevent image-related layout shifts. ' +
```
Comment (pkozlowski-opensource) [inline packages/common/test/directives/ng_optimized_image_spec.ts:215]: nit: the sentence starting with "Including" seems to be duplicated in 2 error messages. Maybe we should pull it into a shared const or create a utility function to report those errors?

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM overall, left a nit comment. Also, please address @AndrewKushnir comments.

File: packages/common/test/directives/ng_optimized_image_spec.ts (hunk 3 above)
Comment (khempenius) [reply packages/common/test/directives/ng_optimized_image_spec.ts:215]: I left those sentences as is for now - I thought that standardizing our messages re: fixes might be good to assess & implement as part of the re-review of the error message review.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
@khempenius I've noticed that the `lint` CI check was failing due to the old base commit in the `image-directive` branch, so I've rebased the `image-directive` branch on top of the most recent version of the main branch. Could you please try to rebase your PR on top of the up-to-date `image-directive` branch? That should help resolve the CI issue. Thank you.

**Review by AndrewKushnir — APPROVED** [review]

**Pull request merged**